packages/teams-js/src/public/thirdPartyCloudStorage.ts

```diff
@@ -0,0 +1,221 @@
+ * actions like drag and drop to compose
+ * @beta
+ */
+export namespace thirdPartyCloudStorage {
+  /** Get context callback function type */
+  let files: FileFor3PApps[] = [];
+  let helper: AttachmentListHelper = {
+    fileType: '',
+    assembleAttachment: [],
+  };
+
+  /**
+   * Object used to represent a file
+   * @beta
+   *
+   */
+  export interface FileFor3PApps extends Blob {
+    /**
+     * A number that represents the number of milliseconds since the Unix epoch
+     */
+    lastModified: number;
+    /**
+     * Name of the file
+     */
+    name: string;
+    /**
+     * file type
+     */
+    type: string;
+    /**
+     * A string containing the path of the file relative to the ancestor directory the user selected
+     */
+    webkitRelativePath?: string;
+  }
+
+  /**
+   * File chunks an output of getDragAndDropFiles API from platform
+   * @beta
+   */
+  export interface FileChunk {
+    /**
+     * Base 64 data for the requested uri
+     */
+    chunk: string;
+    /**
+     * chunk sequence number
+     */
+    chunkSequence: number;
+  }
+
+  /**
+   * Output of getDragAndDropFiles API from platform
+   * @beta
+   */
+  export interface FileResult {
+    /**
+     * Error encountered in getDragAndDropFiles API
+     */
+    error?: SdkError;
+    /**
+     * File chunk which will be assemebled and converted into a blob
+     */
+    fileChunk: FileChunk;
+    /**
+     * File index of the file for which chunk data is getting recieved
+     */
+    fileIndex: number;
+    /**
+     * File type/MIME type which is getting recieved
+     */
+    fileType: string;
+    /**
+     * Indicates whether this file is the last one in a sequence.
+     */
+    isLastFile: boolean;
+    /**
+     * The name of the file.
+     */
+    fileName: string;
+  }
+
+  /**
+   * Helper object to assembled file chunks
+   * @beta
+   */
+  export interface AssembleAttachment {
+    /** A number representing the sequence of the attachment in the file chunks. */
+    sequence: number;
+    /** A Blob object representing the data of the file chunks. */
+    file: Blob;
+  }
+
+  /**
+   * Helper class for assembling files
+   * @beta
+   */
+  export interface AttachmentListHelper {
+    /** A string representing the MIME type of the file */
+    fileType: string;
+    /** An array of {@link AssembleAttachment | AssembleAttachment} objects representing files to be sent as attachment */
+    assembleAttachment: AssembleAttachment[];
+  }
+
+  /**
+   * Defines the Callback function received from Third Party App
+   * @beta
+   */
+  export interface DragAndDropFileCallback {
+    /** Callback from third party app */
+    (files: FileFor3PApps[], error?: SdkError): void;
+  }
+
+  let callback: DragAndDropFileCallback | null = null;
+
+  /**
+   * Get drag-and-drop files using a callback.
+   *
+   * @param {string} dragAndDropInput - thread id received from the app.
```

Comment:
This might have to be a conversation, because this is still not enough information for an external developer to understand how to call this function. What app are you referring to? How does a developer go about asking for and receiving a "thread id"? Is there some format of threadid that can be validating here?

Reply:
@mahimaMSFT We can update this comment to something which developer can understand. If we include what thread id is and it is supposed to be provided by developer, it'll be helpful.

Reply:
Started a new thread with the suggestion